boards/FSM/Inc/App/App_FsmWorld.h

```diff
@@ -5,12 +5,20 @@
 #include "App_InRangeCheck.h"
 #include "App_SharedHeartbeatMonitor.h"
 #include "App_SharedRgbLedSequence.h"
-#include "App_Brake.h"
 #include "App_SharedSignal.h"
 #include "App_SharedClock.h"
 #include "App_AcceleratorPedals.h"
+#include "App_Brake.h"
+#include "App_Coolant.h"
+#include "App_Steering.h"
+#include "App_Wheels.h"
 
 struct FsmWorld;
+struct AcceleratorPedals;
+struct Brake;
+struct Coolant;
+struct Steering;
+struct Wheels;
```

Comment:
i'm not sure how this compiles without linker errors considering these structs are all redefined here and the headers containing their original definitions are included above. why did you need to make this change?

sidenote: @gtaharaedmonds @pasitlaothamatas why aren't we defining the structs in the source files and using extern in the headers? why include the variable definition directly through the header?

Reply:
This compiles **with errors** if it is not there. Gus asked me to do it, so he probably knows why.

Reply:
@kirons96 the struct is defined in the .c file and its being forward declared in the .h file. The rationale behind this was to hide the struct's contents and only access a given structs content through fxns in the .c file (where it is defined)

Reply:
@Lucien950 (refer to option 2) https://www.learncpp.com/cpp-tutorial/forward-declarations/

Reply:
@Lucien950  BTW I'm Dr. Fauci

Reply:
OMG Dr.Fauci! No way

Reply:
@pasitlaothamatas What are you proposing we do then (with option 2)?

Reply:
@Lucien950 just wanted to explain why you need to declare those to compile w/o errors. the example in option 2 refers to fxns, but the same principle applies to structs